### ISSUE TITLE
🐞 automation workflow error

### ISSUE THREAD

**Issue by camelPhonso** [enBloc-org/kindly#240]
# Expected Behaviour

There are a few workflow files in the codebase intended to help the core team manage the progress of Issues in the project.

Two of these workflows should move Issues between the `Backlog` and the `In Progress` states as collaborators get assigned/unassigned.

# Current Behaviour

The workflows `move-issue-to-backlog.yml` and `move-issue-to-in-progress.yml` both fail with the following error: 
```markdown
`issue` is not a valid event name
```

## Steps to Reproduce

1. Visit the Actions tab in the Repo
2. Select the relevant workflow on the left-side panel
3. View the logs on previous runs

**Comment by mnixo** [comment]
Can I take this one?

**Comment by camelPhonso** [comment]
All yours Bubba!

**Comment by mnixo** [comment]
This one ended up being a bit of a rabbit hole. Unfortunately I couldn't find a good solution for it, but at least I'll leave here all the steps that I took.

## Can we just fix the event name?

The (starting) error message is pretty clear. `issue` is not a valid GitHub Actions event type. This probably came from [the example provided for the action being used](https://github.com/kin/gh-action-move-issues-to-column/blob/3837410c959b15a734858d0e1d3a8d05a4b1a212/README.md?plain=1#L10). Using `issues` should fix the issue, [but it doesn't](https://github.com/mnixo/kindly/actions/runs/9422698705/job/25959540970):
```
Error: Target column does not exist on project. Please use a different column name
```
My fork environment seems to replicate the original one just fine: public repository and private project.
So no, we can't just fix the event name.

## Can we just use a different action?

This action seems outdated and doesn't seem to be widely adopted, we could try to use a different one.
I found an official GitHub article: [Moving assigned issues on projects (classic)](https://docs.github.com/en/actions/managing-issues-and-pull-requests/moving-assigned-issues-on-project-boards). It uses a different action: [alex-page/github-project-automation-plus](https://github.com/alex-page/github-project-automation-plus). Let's try it. [Another issue](https://github.com/mnixo/kindly/actions/runs/9423022080/job/25960549352):
```
Error: Could not find the column "In Progress" or project "Kindly (enBloc)"
```
I've set up a dedicated PAT for this (as described in the doc), so this shouldn't be happening. I found [a fairly new issue of the action](https://www.github.com/alex-page/github-project-automation-plus/issues/93) that describes this same problem. It seems that the action is no longer compatible with the GitHub API. The owner is considering archiving the action, stating:
>A lot of the functionality is already built into [GitHubs UI with projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/automating-your-project).

I also found that [GitHub just announced the sunsetting of the Projects (classic)](https://github.blog/changelog/2024-05-23-sunset-notice-projects-classic/) feature. This might explain why actions like these have stopped working.
So no, we can't just use a different action.

## Can we just use the GitHub UI Project automations instead?

There are a few built-in automations that can be used, with most of them being fairly customizable. Unfortunately I couldn't find one that would fit this use case.
So no, we can't just use the GitHub UI Project automations instead?

## Is there some API we can use directly to make this happen?

The only Projects API described in [GitHub's REST API documentation seems to be Projects (classic)](https://docs.github.com/en/rest/projects/projects?apiVersion=2022-11-28), which has been sunset. On the other hand, the GraphQL API shows [ProjectV2](https://docs.github.com/en/graphql/reference/objects#projectv2) objects.
After playing a bit with [GitHub's GraphQL Explorer](https://docs.github.com/en/graphql/overview/explorer), I still couldn't get the column information for an issue in a view of a project.
<details>
<summary>My Sample Query</summary>

```
query {
  repository(owner:"mnixo", name:"kindly") {
    projectsV2(first: 10) {
      nodes {
        title
        items(first: 10) {
          nodes {
            id
            type
            createdAt
            creator {
              login
            }
            project {
              id
            }
            type
            updatedAt
          }
        }
        views(first: 10) {
          nodes {
            name
          }
        }
      }
    }
  }
```

</details>

<details>
<summary>My Sample Query Results</summary>

```
{
  "data": {
    "repository": {
      "projectsV2": {
        "nodes": [
          {
            "title": "Kindly (enBloc)",
            "items": {
              "nodes": [
                {
                  "id": "PVTI_lAHOAEBxCM4Ai8yKzgP4aHM",
                  "type": "ISSUE",
                  "createdAt": "2024-06-07T20:14:22Z",
                  "creator": {
                    "login": "mnixo"
                  },
                  "project": {
                    "id": "PVT_kwHOAEBxCM4Ai8yK"
                  },
                  "updatedAt": "2024-06-07T21:33:36Z"
                }
              ]
            },
            "views": {
              "nodes": [
                {
                  "name": "View 1"
                }
              ]
            }
          }
        ]
      }
    }
  }
}
```

</details>

I couldn't get there myself, but I also found [this other issue describing that the ProjectsV2 GraphQL API can't be used to move an issue to a different column](https://www.github.com/orgs/community/discussions/44265) anyway.
So no, I couldn't find some API that we can use directly to make this happen.

## TL;DR

I tried a few different options but nothing worked out. GitHub is moving from classic to V2 Project APIs and the parity (for what we need) doesn't seem to be there yet. Maybe we'll get more APIs and actions at some point.
Any suggestions are welcome. If nothing pops up, it'll be probably wise to drop this project management shortcut for now.

**Comment by camelPhonso** [comment]
Thanks for sharing the notes from your spike @mnixo 

Is it worth adding the `help wanted` label or marking this as blocked for a few days and see if anyone spots it and can point us to a different option? 

This one is not strictly necessary to the product, it's just to give the core team a better experience of moving issues along the board as people pick them up, so if we really come up with nothing we can remove the workflow entirely and worry later about any other options if needed

**Comment by mnixo** [comment]
I believe it's worth keeping this one open for now, hopefully someone else can make use of my investigation and come up with an alternative approach.

For the time being both workflows are definitely broken. Perhaps we can just disable them in the GitHub UI to prevent additional failed runs.

**Comment by camelPhonso** [comment]
> I believe it's worth keeping this one open for now, hopefully someone else can make use of my investigation and come up with an alternative approach.
> 
> For the time being both workflows are definitely broken. Perhaps we can just disable them in the GitHub UI to prevent additional failed runs.

Agreed. I've disabled them now and am marking this issue as needing help to see if we can prop other ideas 👍
